app/src/main/java/com/epfl/drawyourpath/preferences/ModifyPasswordFragment.kt

```diff
@@ -1,8 +1,74 @@
 package com.epfl.drawyourpath.preferences
 
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
 import androidx.fragment.app.Fragment
 import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.authentication.Auth
+import com.epfl.drawyourpath.authentication.FirebaseAuth
+import com.epfl.drawyourpath.authentication.createAuth
 
 class ModifyPasswordFragment : Fragment(R.layout.fragment_modify_password) {
-    //TODO: add a way to modify the password by writing the old one and the new one 2x.
+
+    private lateinit var auth: Auth
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        auth = createAuth(arguments ?: activity?.intent?.extras)
+
+        view.findViewById<Button>(R.id.BT_Apply).setOnClickListener {
+            onChangePasswordButtonPressed()
+        }
+    }
+
+    /**
+     * Called when the user presses on `APPLY` button.
+     */
+    private fun onChangePasswordButtonPressed() {
+        if (!repeatPasswordMatches()) {
+            showResult("Confirmation password doesn't match")
+            return
```

Comment:
Suggestion: It would be good to avoid hard-coded strings. Better extract them in the strings.xml.

Reply:
Nice remark, I've extracted the strings to the `string.xml` file.